server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -216,6 +225,25 @@ protected void doInternalExecute(
         );
     }
 
+    /**
+     * Track the number of index requests in our APM metrics. We'll track almost all docs here (pipeline or no pipeline,
+     * failure store or original), but some docs don't reach this place (dropped and rejected docs), so we increment for those docs in
+     * different places.
```

Comment:
Ideally, we'd increment the `total` counter in one place, but I couldn't come up with a way to make that happen (because of the reroute vs. dropping "contradiction").